ui/desktop/tests/e2e/authentication.spec.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+/* eslint-disable no-undef */
+const { test } = require('./fixtures/baseTest');
+const LoginPage = require('./pages/loginPage');
+const { expect } = require('@playwright/test');
+
+/* eslint-disable no-undef */
+test.describe('user/password authentication tests', async () => {
+  test('Authenticates and signs out', async ({
+    electronPage,
+    clusterUrl,
+    username,
+    password,
+  }) => {
+    const loginPage = new LoginPage(electronPage);
+    await loginPage.setClusterUrl(clusterUrl);
+    await loginPage.loginWithPassword(username, password);
+
+    await expect(electronPage).toHaveURL(
+      /.*\/scopes\/global\/projects\/targets$/,
+    );
+
+    await electronPage
+      .locator('details')
+      .filter({ hasText: 'Sign Out' })
+      .click();
+
+    await electronPage.getByRole('button', { name: 'Sign Out' }).click();
+    await expect(electronPage).toHaveURL(/.*\/scopes\/global\/authenticate\//);
+  });
+
+  test('Fails to authenticate with wrong password', async ({
+    electronPage,
+    clusterUrl,
+    username,
+  }) => {
+    const loginPage = new LoginPage(electronPage);
+    await loginPage.setClusterUrl(clusterUrl);
+    await loginPage.loginWithPassword(username, 'wrong password!!');
+
+    await expect(electronPage).toHaveURL(/.*\/scopes\/global\/authenticate\//);
+    await expect(electronPage.getByText('Authentication Failed')).toBeVisible();
+  });
+});
+
+test.describe('LDAP authentication tests', async () => {
+  test('Authenticates and signs out', async ({
+    electronPage,
+    clusterUrl,
+    username,
+    password,
+  }) => {
+    const loginPage = new LoginPage(electronPage);
+    await loginPage.setClusterUrl(clusterUrl);
+    await loginPage.loginWithLDAP(username, password);
+
+    await expect(electronPage).toHaveURL(
+      /.*\/scopes\/global\/projects\/targets$/,
+    );
+
+    await electronPage
+      .locator('details')
+      .filter({ hasText: 'Sign Out' })
+      .click();
+
+    await electronPage.getByRole('button', { name: 'Sign Out' }).click();
+    await expect(electronPage).toHaveURL(/.*\/scopes\/global\/authenticate\//);
+  });
+
+  test('Fails to authenticate with wrong password', async ({
```

Comment:
Is this test actually needed since `loginWithLDAP` is calling `loginWithPassword`?

Reply:
I'm not sure I quite understand the issue on why it wouldn't be needed. This test is to test authenticating with LDAP, which also happens to use a user and password to login (but is a different auth method). I could've not called `loginWithPassword` and keep that "only" for user/password login by duplicating the code but I didn't think it would ever change and would most likely affect both if it did.

Whether we login with OIDC, LDAP, or user/pass, we would still need to login with a username/password at the end of the day. Ideally we would also test OIDC but we have some limitations in being able to test that.

Reply:
ok never mind my original question. With the added context careful reading, I see the issue with my original thought and question. Please ignore 😆